### PR TITLE
issue-230 - adjust to `category`-less path structure

### LINKED ISSUE #230: Scripts/blocks adjust to `category`-less path structure
- https://github.com/hlxsites/famous-smoke-cigaradvisor/blob/64fef85be4e65b99868367127e95d394d466cc84/cigaradvisor/scripts/linking-data.js#L89
- https://github.com/hlxsites/famous-smoke-cigaradvisor/blob/64fef85be4e65b99868367127e95d394d466cc84/cigaradvisor/scripts/templates/blog-post.js#L74
- https://github.com/hlxsites/famous-smoke-cigaradvisor/blob/64fef85be4e65b99868367127e95d394d466cc84/cigaradvisor/blocks/article-list/article-list.js#L116

### PR DESCRIPTION
**Pull request by tmaret**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #230

Test URLs:
- Before: https://main--famous-smoke-cigaradvisor--hlxsites.hlx.page/cigaradvisor
- After: https://issue-230--famous-smoke-cigaradvisor--hlxsites.hlx.page/cigaradvisor

### CODE REVIEW

File: cigaradvisor/scripts/linking-data.js [hunk 1]
```diff
@@ -86,7 +86,7 @@ export default function addLinkingData() {
   addOrg(document.querySelector('head'));
   if (window.location.pathname === '/cigaradvisor') {
     addBlogPosts();
-  } else if (window.location.pathname.match(/\/cigaradvisor\/category\/[^/]+$/)) {
+  } else if (window.location.pathname.match(/\/cigaradvisor\/[^/]+$/)) {
```
Comment (bstopp) [inline cigaradvisor/scripts/linking-data.js:89]: I don't think we can just assume this is true. Category paths were known to be below category. Now there are pages at this level of the content hierarchy which are not category pages. We likely need a constant hard-coded list of categories and their paths.

**Review by bstopp — CHANGES_REQUESTED** [review]

File: cigaradvisor/scripts/linking-data.js (hunk 1 above)
Comment (tmaret) [reply cigaradvisor/scripts/linking-data.js:89]: Good point.

> We likely need a constant hard-coded list of categories and their paths.

In `utils.js` or better ?

**Review by Kronnox — APPROVED** [review]

File: cigaradvisor/scripts/templates/blog-post.js [hunk 2]
```diff
@@ -70,9 +71,7 @@ export default async function decorate(main) {
   const articleBlurb = getMetadata('articleblurb');
 
   if (!category) {
-    const url = window.location.href;
-    const regex = /\/cigaradvisor\/category\/([^/]+)/;
-    const match = url.match(regex);
+    const match = getCategory(window.location.href);
```
Comment (kailasnadh790) [inline cigaradvisor/scripts/templates/blog-post.js:74]: can you review this logic: 
image

**Review by kailasnadh790 — CHANGES_REQUESTED** [review]

File: cigaradvisor/scripts/templates/blog-post.js (hunk 2 above)
Comment (tmaret) [reply cigaradvisor/scripts/templates/blog-post.js:74]: @kailasnadh790 good catch ! What's the id of the default category ?

Comment (tmaret) [reply cigaradvisor/scripts/templates/blog-post.js:74]: Ah, `uncategorized` will push a fix

**Comment by tmaret** [comment]
@kailasnadh790 the template path needed to b adjusted to the new path structure (was not loaded)

The remaining issue `Undefined` category in the header comes from the fact that the `loadPosts` method loads the posts from `/cigaradvisor/posts-index.json` and the JSON contains the old paths.

**Review by kailasnadh790 — APPROVED** [review]

**Pull request merged**